### ISSUE TITLE
UI Inventory group search not returning all matching groups

### ISSUE THREAD

**Issue by orgito** [ansible/awx#2002]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - UI
- API

##### SUMMARY

Group search of a dynamic inventory does not return all matching groups. 

##### ENVIRONMENT
* AWX version: 1.0.6.16
* AWX install method: docker on linux
* Ansible version:  2.5.5
* Operating System: Centos 7.5
* Web Browser: Chrome 67

##### STEPS TO REPRODUCE
I search for groups containing "cisco"

##### EXPECTED RESULTS
Return all the matching groups. In that specific case:
- cisco
- cisco_routers
- cisco_switches
- cisco_firewalls
- cisco_other

##### ACTUAL RESULTS

Returns only the group cisco_firewalls

##### ADDITIONAL INFORMATION

The inventory in question is a dynamic inventory. It contains 5568 groups and 609 groups. I used `tower-cli` to confirm that the mentioned groups actually exist. 

Playbooks that I run against those groups will work against the correct host. So the problem seems to be restricted to that functionality.

The same problem does not happen with my static inventories, but they do not contain nearly as many hosts and groups.

**Comment by orgito** [comment]
Looks like the problem is with the API. 

``` bash
$ curl -u admin:admin 'http://awx/api/v2/inventories/4/root_groups/?page_size=20&order_by=name&search=cisco'
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
...
      "name": "cisco_firewalls",
...

$ curl -u admin:admin 'http://awx/api/v2/inventories/4/root_groups/?page_size=20&order_by=name&search=cisco_routers'
{
  "count": 0,
  "next": null,
  "previous": null,
  "results": []
}

$ curl -u admin:admin 'http://awx/api/v2/inventories/4/root_groups/?page_size=20&order_by=name&search=cisco_switches'
{
  "count": 0,
  "next": null,
  "previous": null,
  "results": []
}
```

Existing groups:
- cisco
- cisco_firewalls
- cisco_routers
- csico_switches
- cisco_other

`cisco_*` are children of the `cisco group`

**Comment by ryanpetrello** [comment]
@orgito we've just released 1.0.7, which we believe resolves the underlying issue here. You can try it out here: https://github.com/ansible/awx/releases/tag/1.0.7

Let us know if you're still seeing this issue after installing the latest awx - thanks!
